tests/test_ssl_cert_provider.cpp

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2018 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "src/cert/ssl_cert_provider.h"
+
+#include <gmock/gmock.h>
+
+namespace mp = multipass;
+using namespace testing;
+
+TEST(SSLCertProvider, creates_cert)
```

Comment:
Related to above - what if test machine doesn't have enough entropy?

Reply:
As pointed above, it could block if entropy pool is not initialized. 

In practice, we have not hit this, as test_ssh_key_provider does something similar.